### ISSUE TITLE
Performance of pretrained models

### ISSUE THREAD

**Issue by xinge008** [mit-han-lab/e3d#7]
Hi, thanks for your great work.

You have released the pretrained model, namely, semantickitti_val_spvnas@65gmacs, which achieves miou=64.7 on validation set. How about its performance on test set? 
In fact, I generate the predictions (using evaluate.py with two line output-saving code) on testset based on your code and this pretrained model, however, it achieves miou=60.96. Is it right? Or there are some special techniques. Thanks very much.

**Comment by kentangSJTU** [comment]
Hi,

Thanks for your interest! The major purpose of our current model release is for quick comparison on the validation split since the number of test submissions is limited on the SemanticKITTI benchmark. There're multiple differences between our test submission and the released model on validation set, which causes big difference in performance:

- Our SPVNAS super network is trained on the entire training set and searched on the validation set, then the resulting network is fine-tuned on training + validation. However, the released model is obtained in a different way: the SPVNAS super network is trained on a mini-training set (entire training set - holdout mini-validation set), and searched on the holdout mini-validation set, fine-tuned on the entire training set. 

- When we submit all the models (MinkUNet, SPVCNN, SPVNAS) to the SemanticKITTI test server, we run five-view rotation augmentation, following Choy et al.'s convention on the ScanNet dataset.

These two reasons should be able to explain the results you get, and we are work on releasing the entire pipeline (training + searching + finetuning) soon.



Best,
Haotian

**Comment by xinge008** [comment]
Hi, Haotian,

Thanks for your reply. 

Can you specify the performance gain of these two parts, respectively. Because the current performance on test set has a big gap, I want to make it clear. Thanks very much.

**Comment by kentangSJTU** [comment]
Hi Xing,

Sorry for the delayed response. As I have mentioned, I haven't tried submit models trained only on the training split or models without rotation augmentation to the test server because of the limited number of trials. However, according to my experience on the validation set, I believe the rotation augmentation will bring about 1.5-2.0 mIoU improvement. Hope that information can help you.


Best,
Haotian

**Comment by xinge008** [comment]
Hi, Haotian,

Thanks for your reply.

I have tried the training+validation finetuning, and eight-rotation augmentation. Both of them deliver about 2.5 points improvement, which is far from 6.0 points improvement of your techniques. I am wondering whether there are some other tricks during submitting to test server. Thanks very much.

**Comment by kentangSJTU** [comment]
Hi,

I believe it might be more clear if I restate the pipeline we used for test server submission:

- First, train the SPVNAS super network on the entire training split (seq 00-07, 09-10). The details can be found in our paper Appendix A.1.2.

- Second, search for the best SPVNAS model under MACs constraint on the validation split (seq 08) using evolutionary architecture search.

- Third, finetune the searched SPVNAS model for 10 epochs on the training+validation set (seq 00-10).

- Fourth, run rotation augmentation and submit.

We are still cleaning the code for super network training and architecture search. However, I will be releasing the pretrained SPVNAS super network and searched architectures for submission very soon (hopefully within one week). You may have a try at that time. Hope that the current explanation can help you better understand our pipeline. Thanks!


Best,
Haotian

**Comment by DavideCoppola97** [comment]
@xinge008  Hi, how did you manage to generate the predictions (and related label files) on the test set?
